app/assets/javascripts/discourse/app/instance-initializers/set-custom-ajax-request-headers.js

```diff
@@ -0,0 +1,19 @@
+import { setCustomHeader } from "discourse/lib/ajax";
+import PreloadStore from "discourse/lib/preload-store";
+
+export default {
```

Comment:
I tried to keep this initializer as generic as possible but it does add a small cost since this initializer has to be present in all environments. 

cc @davidtaylorhq I was wondering if you might have a better idea here.